lib/compiler/src/engine/artifact.rs

```diff
@@ -509,14 +496,49 @@ impl Artifact {
             .map(|_function_body| 0u64)
             .collect::<PrimaryMap<LocalFunctionIndex, u64>>();
 
-        let mut metadata = ModuleMetadata {
+        let metadata = ModuleMetadata {
             compile_info,
             prefix: prefixer.as_ref().map(|p| p(data)).unwrap_or_default(),
             data_initializers,
             function_body_lengths,
             cpu_features: target.cpu_features().as_u64(),
         };
 
+        Ok((metadata, module_translation, function_body_inputs))
+    }
+
+    /// Compile a module into an object file, which can be statically linked against.
+    ///
+    /// The `prefixer` returns the a String to prefix each of the
+    /// functions in the static object generated by the
+    /// so we can assure no collisions.
+    #[cfg(feature = "static-artifact-create")]
+    pub fn generate_object<'data>(
+        compiler: &dyn Compiler,
+        data: &[u8],
+        object_name: &str,
```

Comment:
what is object name exactly? We should probably use prefixer as well for the metadata symbol

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3430/commits/253cd2118383357880c26b83bbd84325d49543c8